determined-ai/determined
# fix: add user modal should not disable submit button (DET-10277)

## Ticket

[DET-10277]

## Description

Fixes an issue where the submit button is only disabled/enabled on page (re)load, causing the submit button to become permanently disabled if any form errors are registered when the user changes tabs (other behaviors might produce the same effect, the one we consistently reproduced the issue with was changing browser tabs).

This seems like a bug that would've been around for almost a year, maybe longer, but it's more noticeable now that there are more fields after the first that could have validation errors and give admins reasons to change tabs while the modal is still open.

## Test Plan

1. Log in with an admin user.
2. Navigate to Admin Settings.
3. Click "Add User".
4. Click "Submit" with an empty form, or with errors appearing on the form.
5. Open a new tab (doesn't have to load another page).
6. Return to the tab with the open form.
7. The "Save" button should not be disabled.
8. Correcting any form errors should make the displayed errors/warnings go away.
9. Creating a user should succeed if no errors are showing on screen.

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
